pytorch/matmul.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+# Copyright(C) 2019 Francesco Murdaca
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Performance Indicator (PI): matmul for PyTorch (Thoth Team)."""
+
+import logging
+import os
+import sys
+import numpy as np
+import json
+from timeit import time
+
+import torch
+
+_LOGGER = logging.getLogger(__name__)
+
+_ARGS_DTYPE_MAP = {
+    "32-bit floating point": {
```

Comment:
Can we have these named as in case of TensorFlow performance indicators? (`float32`, ...)

Reply:
PyTorch has more than the one in TensorFlow, so only some of them could be matched.

Reply:
Let's share what they have in common and introduce a new naming where needed. I can imagine a query where we would be comparing TF and PyTorch performance for a specific bit width respecting model used (e.g. optimized models which reduce bit size). IMHO no need to have two names for same thing.

Reply:
Agreed :) I updated values, matching with TF for the one compatible. WDYT?

Reply:
```
    "8-bit integer (unsigned)": {
        "dtype": torch.uint8,
    },
    "8-bit integer (signed)": {
        "dtype": torch.int8,
    },
    "16-bit integer (signed)": {
        "dtype": torch.int16 or torch.short,
    },
    "64-bit integer (signed)": {
        "dtype": torch.int64 or torch.long,
    },
```

Reply:
These ones could change to `uint8`, `int8`, `int16` and `int64`